### ISSUE TITLE
Enable reservation

### ISSUE THREAD

**Issue by SanderBorgmans** [pyiron/pysqa#39]
To couple a reservation code to a job in pyiron, I am using the following function:

```
def enable_reservation_command(self, process_id,reservation_id):
        return ["scontrol", "update", "job", str(process_id), "reservation={}".format(reservation_id)]
```
However, due to a bug(?) in slurm, the reservation_id can be assigned to the job, but its priority is not bumped. As such, it is required to set the $SBATCH_RESERVATION environment variable right before calling sbatch to have an increased priority. To do this, there are two options:

- assign it before the sbatch command is called in bash or python
     > export SBATCH_RESERVATION=name-of-reservation
     > os.environ['SBATCH_RESERVATION'] = 'name-of-reservation'
- use the reservation flag when using sbatch
    > sbatch --reservation=name-of-reservation ...

Would there be an easy way of doing this?

**Comment by jan-janssen** [comment]
I am sorry I do not use reservations regularly, is `reservation_id` the same as `name-of-reservation`? If that is the case it might be easier to have all commands in the same `enable_reservation_command` function. In addition it might be even more generic to have these commands available as shell scripts as well, where the user can write their own shell scripts for these commands and we only provide templates.

**Comment by SanderBorgmans** [comment]
@jan-janssen the `reservation_id` is indeed the same as `name-of-reservation`. I was indeed also thinking about some sort of decorater function, which could just decorate all the commands and add the extra flag when it is provided. Generic shell scripts could work for sure as well. Depends on how you envision the pysqa module to evolve. I'll try something with a decorater function on my forked branch then, I'll keep you posted.

**Comment by jan-janssen** [comment]
Perfect - thanks a lot. I guess currently there is not yet a clear strategy all I learned so far is that while queuing systems offer a lot of settings but most users only use a few. Therefore we came up with the idea to use jinja templates based on the shell scripts most HPC computing centres provide to make it as easy as possible for the user to configure their queuing system. I guess using the shell scripts for the queue adapters as well is a logical extension.
